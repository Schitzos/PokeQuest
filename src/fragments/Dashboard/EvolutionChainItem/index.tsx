import * as React from 'react';
import {StyleSheet} from 'react-native';
import {View} from 'react-native';
import FastImage from 'react-native-fast-image';
import TextView from '@/components/TextView';
import theme from '@/theme';
import Button from '@/components/Button';

interface ItemProps {
  id: number;
  name: string;
  weight: number | undefined;
  handleChooseEvolution: (e: number) => void;
}
const EvolutionChainItem: React.FC<ItemProps> = ({
  id,
  name,
  weight,
  handleChooseEvolution,
}) => {
  return (
    <View style={styles.base}>
      <FastImage
        source={{
          uri: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${id}.png`,
          priority: FastImage.priority.high,
        }}
        style={styles.image}
        defaultSource={require('@assets/images/default_image_loading.png')}
        resizeMode={FastImage.resizeMode.contain}
      />
      <TextView fz={20} color={theme.colors.white}>
        {name}
      </TextView>
      <TextView fz={14} color={theme.colors.white}>
        Weight Needed "{weight}"
      </TextView>
      <Button onPress={() => handleChooseEvolution(id)}>Choose {name}</Button>
    </View>
  );
};

export default EvolutionChainItem;

const styles = StyleSheet.create({
  base: {
    flex: 1,
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center',
    height: 500,
  },
  image: {
    width: 225,
    height: 225,
  },
});
