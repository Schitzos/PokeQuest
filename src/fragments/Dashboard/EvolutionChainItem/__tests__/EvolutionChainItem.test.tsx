import React from 'react';
import {render} from '@testing-library/react-native';
import EvolutionChainItem from '../index';

describe('EvolutionChainItem component', () => {
  const itemProps = {
    id: 1,
    name: 'Bulbasaur',
    weight: 20,
    handleChooseEvolution: jest.fn(),
  };

  test('renders correctly', () => {
    const {getByText} = render(<EvolutionChainItem {...itemProps} />);
    expect(getByText('Bulbasaur')).toBeTruthy();
    expect(getByText('Weight Needed "20"')).toBeTruthy();
  });

  test('matches snapshot', () => {
    const {toJSON} = render(<EvolutionChainItem {...itemProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
