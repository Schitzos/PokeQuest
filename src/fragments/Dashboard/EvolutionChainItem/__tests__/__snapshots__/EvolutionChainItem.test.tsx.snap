// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvolutionChainItem component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 500,
      "justifyContent": "center",
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 225,
          "width": 225,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={
        {
          "testUri": "../../../assets/images/default_image_loading.png",
        }
      }
      resizeMode="contain"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "PixeloidMono",
          "fontSize": 20,
          "fontWeight": "normal",
          "textAlign": "left",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    Bulbasaur
  </Text>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "PixeloidMono",
          "fontSize": 14,
          "fontWeight": "normal",
          "textAlign": "left",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    Weight Needed "
    20
    "
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#00AFF0",
        "borderRadius": 8,
        "flex": 0,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "PixeloidMono",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Choose 
      Bulbasaur
    </Text>
  </View>
</View>
`;
